sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -1942,3 +1963,94 @@ def __getitem__(self, snapshot_id: SnapshotId) -> Snapshot:
         if snapshot is None:
             raise KeyError(snapshot_id)
         return snapshot
+
+
+class SharedVersionSnapshot(PydanticModel):
```

Comment:
should this go in snapshot definition?

Reply:
It's very specific to the State Sync implementation and is only used internally